postcommit.py

```diff
@@ -0,0 +1,642 @@
+#       *************************************************************************************************************** 
+
+    	def getDiffContent(self, changeset1, changeset2):
+        	diff_command = "git diff -U0 " + changeset1 + " " + changeset2 
+        	result = status_check(diff_command, self.module_dir)
+        	diff_changes = self.parseDiffContent(list(result))
+        	return diff_changes
+
+
+#	***************************************************************************************************************
+#	getDiffLines :
+#
+#			Find number of lines newly added in file
+#
+#			@param : changes
+#
+#			@return : [1,2,4,5]
+#       ***************************************************************************************************************
+    	def getDiffLines(self,changes):
+        	difflines = []
+        	for val in changes:
+            		changed_line_values = str(val)
+
+            	if changed_line_values.find(',') > -1:
+                	start_linenumber = int(changed_line_values.split(',')[0])
+                	total_changed_lines = int(changed_line_values.split(',')[1])
+                	for line in range(start_linenumber, start_linenumber + total_changed_lines):
+                    		difflines.append(line)
+            	else:
+                	difflines.append(int(changed_line_values))
+
+        	return difflines
+
+#       ***************************************************************************************************************
+#	parseDiffContent :
+#
+#			Parse diff contents for two changesets
+#
+#			@param : diff
+#			
+#			@return : [{'changed_lines': [1,4,6], 'file': 'test'}]
+#       *************************************************************************************************************** 
+    	def parseDiffContent(self, diff):
+        	diff_changes = []
+        	file_name = match_values = []
+        	for diff_content in diff:
+            		file_name_pattern = re.compile('^\\+\\+\\+ b/([^\\s]+)', re.M)
+            		modified_lines_pattern = re.compile('^@@[-0-9,\\s]+\\+([0-9+,]+)\\s*@@', re.M)
+
+            		if not file_name:
+                		file_name = file_name_pattern.findall(diff_content)
+            		if not match_values:
+                		match_values = modified_lines_pattern.findall(diff_content)	
+
+            		if file_name and match_values:
+                		file_name = file_name[0].replace('/', os.sep).strip()
+                		diff_changes.append({'file': file_name, 'changed_lines': self.getDiffLines(match_values)})
+                		file_name = match_values = []
+
+        	return diff_changes
+
+result = {}
+if type == "hg":
+    repository = localrepository(ui(), repopath)
+    hgObj = HGRepository(repository, ui())
+    result['commits'], result['newchangesets'] = hgObj.getCommits()
+
+else:
+    gitObj = GitRepository(repopath)
+    result['commits'], result['newchangesets'] = gitObj.getCommits()
+
+print result
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace print statement by built-in function. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=python%3APrintStatementUsage)